packages/stark-core/src/modules/session/services/session.service.spec.ts

```diff
@@ -578,32 +578,29 @@ describe("Service: StarkSessionService", () => {
 			mockKeepaliveService.request.calls.reset();
 
 			// make sure the fake pre-auth info is set correctly
-			const preAuthDefaults: StarkPreAuthentication = sessionServiceHelper.getFakePreAuthenticationDefaults();
-
-			const expectedPreAuthHeaders: object = {};
-			expectedPreAuthHeaders[StarkHttpHeaders.NBB_USER_NAME] = mockUser.username;
-			expectedPreAuthHeaders[StarkHttpHeaders.NBB_FIRST_NAME] = mockUser.firstName;
-			expectedPreAuthHeaders[StarkHttpHeaders.NBB_LAST_NAME] = mockUser.lastName;
-			expectedPreAuthHeaders[StarkHttpHeaders.NBB_MAIL] = mockUser.email;
-			expectedPreAuthHeaders[StarkHttpHeaders.NBB_LANGUAGE] = mockUser.language;
-			expectedPreAuthHeaders[StarkHttpHeaders.NBB_DESCRIPTION] =
-				mockUser.referenceNumber + preAuthDefaults.descriptionSeparator + mockUser.workpost;
-			expectedPreAuthHeaders[StarkHttpHeaders.NBB_ROLES] = mockUser.roles.join(preAuthDefaults.roleSeparator);
-
-			sessionServiceHelper.setFakePreAuthenticationHeaders(mockUser);
-
+			const devAuthDefaults: StarkPreAuthentication = sessionService.getDevAuthenticationDefaults();
+			const description: string =
+				devAuthDefaults.defaults.referenceNumber + devAuthDefaults.descriptionSeparator + devAuthDefaults.defaults.workpost;
+
+			const expectedDevAuthHeaders: Map<string, string> = new Map<string, string>();
+			expectedDevAuthHeaders.set("usernameTestHeader", mockUser.username);
+			expectedDevAuthHeaders.set("firstnameTestHeader", mockUser.firstName);
+			expectedDevAuthHeaders.set("lastnameTestHeader", mockUser.lastName);
+			expectedDevAuthHeaders.set("emailTestHeader", <string>mockUser.email);
+			expectedDevAuthHeaders.set("descriptionTestHeader", description);
+			expectedDevAuthHeaders.set("rolesTestHeader", mockUser.roles.join(devAuthDefaults.roleSeparator));
+
+			sessionServiceHelper.setDevAuthenticationHeaders(expectedDevAuthHeaders);
 			sessionServiceHelper.configureKeepaliveService();
 
 			expect(mockKeepaliveService.interval).toHaveBeenCalledTimes(1);
 			expect(mockKeepaliveService.interval).toHaveBeenCalledWith(appConfig.keepAliveInterval);
 			expect(mockKeepaliveService.request).toHaveBeenCalledTimes(1);
 
 			let mockHeadersObj: HttpHeaders = new HttpHeaders();
-			mockHeadersObj = mockHeadersObj.set(StarkHttpHeaders.NBB_CORRELATION_ID, mockCorrelationId);
+			mockHeadersObj = mockHeadersObj.set(mockCorrelationIdHeaderName, mockCorrelationId);
 
-			for (const key of Object.keys(expectedPreAuthHeaders)) {
-				mockHeadersObj = mockHeadersObj.set(key, expectedPreAuthHeaders[key]);
-			}
+			expectedDevAuthHeaders.forEach((value: string, key: string) => (mockHeadersObj = mockHeadersObj.set(key, value)));
```

Comment:
I had to copy the values from the **expectedDevAuthHeaders** object into the **mockHeadersObj** object because the **HttpRequest** requires an object of type **HttpHeaders**